docs/cookbook/creating-webhooks.md

```diff
@@ -0,0 +1,93 @@
+
+<AposCodeBlock>
+
+```javascript
+handlers(self) {
+  return {
+    'afterPublish': {
+      async sendWebhook(req, data) {
+        // data sent by the afterPublish event includes the document data 
+        // and whether it is being published for the first time
+        // https://v3.docs.apostrophecms.org/reference/server-events.html#afterpublish
+        if (!data.firstTime) {
+          return;
+        }
+
+        await self.apos.http.post('https://slack-webhook-url-here', {
+          body: {
+            text: 'do something cool with the data'
+          }
+        });
+      }
+    }
+  };
+}
+```
+  <template v-slot:caption>
+    /modules/article/index.js
+    </template>
+
+</AposCodeBlock>
+
+In this block of code, we are using the `handlers(self)` module configuration function. We could use any server event, but in this case we are using the `afterPublication` event that is emitted by our custom article piece-type module. This server event delivers two parameters, `req` and `data`. The `data` parameter contains information about the document being published including the content and whether this is the first time it is being published. In this case, we are adding an early return if this isn't the first time the document is being published.
+
+The next section of code is involved in setting up the HTTP `POST` request using the `post` method of the [`@apostrophecms/http` module](https://v3.docs.apostrophecms.org/reference/modules/http.html#async-post-url-options) to send our data to the endpoint. Depending on the endpoint, the returned response might be as simple as a `200` success or a `400` failed response, or might contain additional data. That response can be handled to retry a failed response or in some way log the returned data.
+
+## Incoming webhooks
+
+Your project may require a webhook endpoint to receive incoming notices from services, like a payment portal or subscription manager. Much like Slack can add a message to a channel when it receives data from your site on the correct endpoint, your site can be set up to make changes to the database or perform other tasks when data is received at a specific endpoint. This can be accomplished by using the Apostrophe [`apiRoutes(self)` module configuration](https://v3.docs.apostrophecms.org/reference/module-api/module-overview.html#apiroutes-self) function. This allows you to easily set up an endpoint for any HTTP request method.
+
+<AposCodeBlock>
+
+```javascript
+options: {
+  csrfExceptions: [ '/webhooks' ]
+},
+apiRoutes(self) {
+  return {
+    post: {
+      // change function name to match the desired endpoint URL
+      '/webhooks': async function(req) {
```

Comment:
As discussed I'm fine with this being a rooted URL.

Reply:
I'm twitching to ask that this be a real and specific integration with a known service, but despite seeing docs for outgoing webhooks in a zillion services, none are really a great match. So I think it's fine as it is.